src/cms/views/pages/page_actions.py

```diff
@@ -112,6 +118,99 @@ def delete_page(request, page_id, region_slug, language_code):
     )
 
 
+# pylint: disable=unused-argument
+def link_callback(uri, rel):
+    """
+    according to xhtml2pdf documentation this function is neccessary for
+    resolving the django static files references and returning absolute paths
+    to the files
+
+    :param uri: uri that is generated by django template tag 'static'
+    :type uri: str
+    :return: the absolute path according to django's static file settings
+    :rtype: str
+    """
+    sUrl = settings.STATIC_URL
+    sRoot = settings.STATIC_ROOT
+    mUrl = settings.MEDIA_URL
+    mRoot = settings.MEDIA_ROOT
+    if uri.startswith(mUrl):
+        path = os.path.join(mRoot, uri.replace(mUrl, ""))
+    elif uri.startswith(sUrl):
+        path = os.path.join(sRoot, uri.replace(sUrl, ""))
+    else:
+        return uri
+    # make sure that file exists
+    if not os.path.isfile(path):
+        raise Exception(
+            "media URI must start with %s or %s, but received uri was %s"
+            % (sUrl, mUrl, uri)
+        )
+    return path
+
+
+@login_required
+@region_permission_required
+# pylint: disable=unused-argument
+def export_pdf(request, region_slug, language_code):
+    """
+    function for handling a pdf export request for a single page
+    if the page is a root page the corresponding django template
+    inserts the content of all children pages
+
+    :param request: request submitted for rendering pdf document
+    :type request: ~django.http.HttpRequest
+    :param language_code: bcp47 code of the current language
+    :type language_code: str
+    :raises PermissionDenied: user login and permissions required
+    :return: pdf document offered for download
+    :rtype: ~django.http.HttpRequest
+    """
+    page_ids = []
+    region = Region.get_current_region(request)
+    for page_id in request.GET.get('pages').split(','):
+        page_ids.append(page_id)
+    # retrieve all selected pages
+    pages = region.pages.filter(archived=False, id__in=page_ids)
+    amount_pages = len(pages)
+    pdf_key_list = [region_slug]
+    for page in pages:
+        # retrieve the translation for each page
+        page_translation = page.get_translation(language_code)
+        pdf_key_list.append(page_translation.id)
+        pdf_key_list.append(page_translation.last_updated)
+    pdf_key_string = '_'.join(map(str, pdf_key_list))
+    pdf_hash_key = hashlib.sha256(bytes(pdf_key_string, 'utf-8')).hexdigest()
+    cached_response = cache.get(pdf_hash_key, 'has_expired')
+    if (cached_response is not 'has_expired'):
+        return cached_response
+    else:
```

Comment:
So basically this is where the caching happens, in the first lines it's just retrieving all queried pages from database. Then in line 182 all page_translation ids and timestamps are combined to a single string and hashed in the following line.